This should allow RequestHeader statements to be passed to a vhost.

Aethylred opened the pull request:
This. can either be a single string or and array of statements.

More detail on RequestHeader declarations can be found here:
http://httpd.apache.org/docs/2.2/mod/mod_headers.html#requestheader

Commit:
This should allow RequestHeader statements to be passed to a vhost. This can either be a single string or and array of statements.

More detail on RequestHeader declarations can be found here:
http://httpd.apache.org/docs/2.2/mod/mod_headers.html#requestheader

Commit:
fixing typos.

Commit:
apache::mod{'headers':} is needed by both RedHat and Debian.

hunner commented on manifests/default_mods.pp:
```diff
@@ -60,4 +59,5 @@
   apache::mod { 'authz_host': }
   apache::mod { 'authz_user': }
   apache::mod { 'env': }
+  apache::mod { 'headers': }
```
I believe the `headers` module is not enabled by default on Debian, which is why it was not included here. I may be wrong though; which version of Debian were you testing this on?

hunner replied:
Perhaps we could create an `apache::mod::headers` class than can conditionally be included inside `apache::vhost` when `request_header` is passed.

Aethylred replied:
I use Ubuntu 12.10, which has `mod_headers` on by default, but I like the suggestion of enabling `apache::mod::headers` (which doesn't exist yet) automatically if `request_headers` is passed.

hunner commented on templates/vhost/_requestheader.erb:
```diff
@@ -0,0 +1,12 @@
+  RequestHeader <%= @request_header %>
+  <% end -%>
+
+<% end -%>
```
You can also ensure `@request_headers` is a non-emptystring array no matter what is passed in like:

``` erb
<% if @request_headers and ! @request_headers.empty? -%>

  ## Request rules
<% Array(@request_headers).each do |request_header| -%>
    RequestHeader <%= request_header %>
<% end -%>
<% end -%>
```

Also remember to update the `apache::vhost` template comment with the new fragment and used variables.

Aethylred replied:
Yes, and should also do nothing if the array contains empty strings `['','']`

...actually the strings should be validated, but I'm not writing a syntax for it now.

hunner commented on manifests/vhost.pp:
```diff
@@ -86,6 +86,7 @@
     $redirect_dest      = undef,
     $redirect_status    = undef,
     $rack_base_uris     = undef,
+    $request_header     = undef,
```
Perhaps this should be `$request_headers` since it could accept an array?

Aethylred replied:
Agreed.

Commit:
request_headers is usually an array, so use a plural variable name

Commit:
Added comments for request_headers

Commit:
Created the class apache::mod::headers to specifically handle mod_headers. This will automatically be enabled if the request_headers parameter is defined when calling apache::vhost

Commit:
_requestheader.erb now rejects empty arrays, and skips empty strings in arrays.

an array of empty strings ['',''] will still include the fragment, but no RequestHeader rules, which should be safe.
